sound/soc/sof/control.c

```diff
@@ -65,6 +65,53 @@ static inline u32 ipc_to_mixer(u32 value, u32 *volume_map, int size)
 	return i - 1;
 }
 
+static void snd_sof_refresh_control(struct snd_sof_control *scontrol)
+{
+	enum sof_ipc_ctrl_type ctrl_type;
+	struct sof_ipc_ctrl_data *cdata;
+	struct snd_soc_component *scomp;
+	int ret;
+
+	if (!scontrol->updated_in_firmware)
+		return;
+
+	cdata = scontrol->control_data;
+	scomp = scontrol->scomp;
+
+	if (scontrol->cmd == SOF_CTRL_CMD_BINARY)
+		ctrl_type = SOF_IPC_COMP_GET_DATA;
+	else
+		ctrl_type = SOF_IPC_COMP_GET_VALUE;
+
+	ret = pm_runtime_get_sync(scomp->dev);
+	if (ret < 0 && ret != -EACCES) {
+		dev_err_ratelimited(scomp->dev,
+				    "error: %s pm_runtime_get_sync failed: %d\n",
+				    __func__, ret);
+		pm_runtime_put_noidle(scomp->dev);
+		return;
```

Comment:
can it happen that you have to resume the device after getting a firmware notification?

I don't fully get the flow here, on a notification userspace does a get and this is to protect against the device suspending in the mean time? If yes, what are the odds that the value is still relevant?

Reply:
I have been hesitant around this.
Most likely we should just:
```
if (!pm_runtime_active(scomp->dev))
	return;
```
But if this is an issue then we already have it in use via `snd_sof_bytes_ext_volatile_get()`, my thinking was that since that code is already there, let's do the same here.